src/www/firewall_rule_lookup.php

```diff
@@ -53,7 +55,22 @@
                         header(url_safe('Location: /%s?if=%s&id=%s', [$parts['path'], $query['if'], $query['id']]));
                     }
                 } else {
-                    header(sprintf('Location: /%s', $rule->getRef()));
+                    // search model firewall rule
+                    $interface = '';
+                    foreach ((new Filter())->rules->rule->iterateItems() as $ruleItem) {
+                        if ($ruleItem->getAttribute('uuid') === $rid) {
+                            $interfaceValue = $ruleItem->interface->getValue();
+                            // multiple interfaces in a rule are skipped as empty string is the default for floating
+                            if ($interfaceValue !== '' && strpos($interfaceValue, ',') === false) {
+                                $interface = $interfaceValue;
+                            }
+                            break;
+                        }
+                    }
+
+                    $base = strtok($rule->getRef(), '#');
+                    $hash = url_safe('interface=%s&edit=%s', [$interface, $rid]);
+                    header("Location: /{$base}#{$hash}");
```

Comment:
if this is a trampoline into the mvc code, it should probably just go there with a uuid, it feels like the interface is a bit irrelevant here.

Reply:
I see your point, in that case this might indeed be the shortest path (combined with a direct fetch of the node)